### PR TITLE
Lazy initial scripts

### PR DESCRIPTION
**Pull request by naveen-variant**
Adding initial code changes . I will update read me file with the inputs required to call this action .

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1-alpine3.12
+
+#install asp.net libraries
+RUN aspnetcore_version=3.1.11 \
```
Comment (vibindaniel) [inline Dockerfile:4]: Doesn't sdk come with the dotnet cli?

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    required: true
+  docker_push:  # docker push
+    description: 'docker push'
+    required: true
+
+
+runs:
+  using: 'docker'
+  image: docker://public.ecr.aws/p4v7w0a5/lazy/donet-base
+  env:
```
Comment (vibindaniel) [inline action.yml:62]: I don't think you need to send env like this. Even if you don't provide the input will come as `INPUT_<variable>`

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
+#!/bin/sh -l
+set -euo pipefail
+
+echo "setting prerequisites."
+export BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/} | tr -d '[:punct:]')
+
+
+if [ $BRANCH_NAME != 'master' ];
```
Comment (vibindaniel) [inline entrypoint.sh:8]: Can you use gitversion to get the version?

File: entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+echo "Current directory : $GITHUB_WORKSPACE"
+
+echo "Start: code build."
+if [ -z "${CUSTOM_DOCKER_FILE_PATH}" ];
+then
+   echo "Running lazy build steps."
+   cd $GITHUB_WORKSPACE && dotnet build --configuration Release
+   echo "Start run tests."
+   cd $GITHUB_WORKSPACE && dotnet test
```
Comment (vibindaniel) [inline entrypoint.sh:26]: sonar scan also needs to do a test. can you merge both?

File: entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+else
+   echo "Running custom build steps."
+   chmod +x /lazy-scripts/build.sh && /lazy-scripts/build.sh ${DOCKER_REPO_NAME} ${DOCKER_IMAGE_NAME} sha-${GITHUB_SHA:0:7}
```
Comment (vibindaniel) [inline entrypoint.sh:30]: Docker build should happen after sonar scan

File: entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+fi
+
+echo "Start: Publish nuget push."
+echo "nugetflag: $NUGET_PUSH"
+if [ $NUGET_PUSH == 'true' ]
+then
+    echo "Start publishing image to github registry."
+    if [ -z "${CUSTOM_DOCKER_FILE_PATH}" ];
+    then
+        echo "lazy flow"
```
Comment (vibindaniel) [inline entrypoint.sh:69]: If this is a todo, please add `Todo`comment

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+RUN apk add --no-cache tzdata ca-certificates
+
+#install docker
+RUN apk update && apk add --no-cache docker-cli
+#install jdk
+RUN apk --no-cache add openjdk11
+
+#install python and awscli
+RUN apk add --no-cache \
```
Comment (vibindaniel) [inline Dockerfile:19]: Please install awscli v2. This install v1

File: lazy-scripts/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,12 @@
+ARG VERSION=3.1-alpine3.10
+FROM mcr.microsoft.com/dotnet/core/sdk:$VERSION 
+
+
+RUN dotnet publish src/ -c Release -o ./output/api
```
Comment (vibindaniel) [inline lazy-scripts/Dockerfile:5]: This seems very specific to a certain repo. Make it generic

File: lazy-scripts/build.sh [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+function cleanup()
+{
+  set +euo pipefail
+  docker image rm ${IMAGE}
+}
+
+trap "cleanup" EXIT
+
+docker build -t ${IMAGE} --build-arg "GITHUB_USER" --build-arg "GITHUB_TOKEN" .
+#curl -s https://ci-tools.anchore.io/inline_scan-latest | bash -s -- -r -f -d Dockerfile ${IMAGE}
```
Comment (vibindaniel) [inline lazy-scripts/build.sh:23]: remove comment

File: lazy-scripts/build.sh [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+function cleanup()
+{
+  set +euo pipefail
+  docker image rm ${IMAGE}
+}
+
+trap "cleanup" EXIT
+
+docker build -t ${IMAGE} --build-arg "GITHUB_USER" --build-arg "GITHUB_TOKEN" .
```
Comment (vibindaniel) [inline lazy-scripts/build.sh:22]: Why is `GITHUB_TOKEN` & `GITHUB_USER` required?

File: lazy-scripts/publish.sh [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+if [ "$#" -ne 5 ]; then
+    echo "Usage: ./publish.sh [AWS_ACCOUNT_ID] [REPO_NAME] [IMAGE_NAME] [IMAGE_TAG] [DOCKERFILE_FULL_PATH]"
+    exit 1
+fi
+
+AWS_ACCOUNT_ID=${1}
+DOCKER_REPO_NAME=${2}
+IMAGE_NAME=${3}
+IMAGE_TAG=${4}
+DOCKER_REPO_HOST=${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com
```
Comment (vibindaniel) [inline lazy-scripts/publish.sh:14]: Get account ID from 
https://docs.aws.amazon.com/cli/latest/reference/sts/get-caller-identity.html

File: lazy-scripts/run_tests_sonarscan.sh [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+  -r, --revision              (Required) SHA of the commit being analyzed to enable PR/commit annotations.
+  -b, --branch-name           Branch name to associate results. Cannot be used with --pull-request-key
+  -p, --pull-request-key      Pull request key to associate results. Cannot be used with --branch-name
+    The environment variable SONAR_TOKEN is used for the sonar.login property
+EOF
+
+  exit "$exitcode"
+}
+
+function process_cli
```
Comment (vibindaniel) [inline lazy-scripts/run_tests_sonarscan.sh:32]: Make the scripts consistent. If you are using cli strategy here, use it in every script

**Review by vibindaniel — CHANGES_REQUESTED** [review]
1. Add worfklow to auto build & lint
2. Update Readme

File: Dockerfile (hunk 1 above)
Comment (naveen-variant) [reply Dockerfile:4]: Yes sdk comes with dotnet CLI.

File: action.yml (hunk 2 above)
Comment (naveen-variant) [reply action.yml:62]: Changed to Input_ variables

File: entrypoint.sh (hunk 3 above)
Comment (naveen-variant) [reply entrypoint.sh:8]: Changed to use gitversion.

File: lazy-scripts/build.sh (hunk 9 above)
Comment (naveen-variant) [reply lazy-scripts/build.sh:23]: Removed .

File: lazy-scripts/build.sh (hunk 10 above)
Comment (naveen-variant) [reply lazy-scripts/build.sh:22]: Development team is using in their Dockerfile and expecting this argument .

File: lazy-scripts/publish.sh (hunk 11 above)
Comment (naveen-variant) [reply lazy-scripts/publish.sh:14]: Changed to use from sts caller indentity .

File: entrypoint.sh (hunk 5 above)
Comment (naveen-variant) [reply entrypoint.sh]: isn't it other way. First we build the code to see if there are any compilation errors or packaging errors and  then proceed to run tests.

File: lazy-scripts/run_tests_sonarscan.sh (hunk 12 above)
Comment (naveen-variant) [reply lazy-scripts/run_tests_sonarscan.sh:32]: I will make everything consistent as we keep adding lazy flow.

File: lazy-scripts/Dockerfile (hunk 8 above)
Comment (naveen-variant) [reply lazy-scripts/Dockerfile]: Updated dockerfile . Still need some corrections as we start building lazy flow.

File: entrypoint.sh (hunk 5 above)
Comment (vibindaniel) [reply entrypoint.sh]: Building code is different from docker build. In docker build, you are building the image.

File: lazy-scripts/build.sh (hunk 10 above)
Comment (vibindaniel) [reply lazy-scripts/build.sh:22]: Can you direct me to where developer team is using it?

File: entrypoint.sh (hunk 4 above)
Comment (naveen-variant) [reply entrypoint.sh]: Updated

File: entrypoint.sh (hunk 5 above)
Comment (naveen-variant) [reply entrypoint.sh]: Moved building image after sonar scan.

File: lazy-scripts/run_tests_sonarscan.sh (hunk 12 above)
Comment (vibindaniel) [reply lazy-scripts/run_tests_sonarscan.sh:32]: If you aren't doing now, then create a tech debt ticket to track it.

File: lazy-scripts/build.sh (hunk 10 above)
Comment (vibindaniel) [reply lazy-scripts/build.sh:22]: This hasn't been done yet

File: Dockerfile (hunk 1 above)
Comment (vibindaniel) [reply Dockerfile]: Then why do you have to install aspnetcore?

File: README.md [hunk 13]
```diff
@@ -1,2 +1,168 @@
...
+
+```
+
+## parameters
+
+### Input Parameters
+
+Input parameters section.
+
+```text
```
Comment (vibindaniel) [inline README.md:70]: Don't use code-block. Instead, use normal headers. It is unreadable now.
https://github.com/variant-inc/lazy-action-dotnet/tree/lazy-initial-scripts

File: lazy-scripts/run_tests_sonarscan.sh [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+    usage 1
+fi
+
+function sonar_logout()
+{
+  set +eu
+  dotnet sonarscanner end /d:sonar.login=${SONAR_TOKEN}
+}
+
+if [ ! -z ${BRANCH_NAME} ]; then
```
Comment (vibindaniel) [inline lazy-scripts/run_tests_sonarscan.sh:95]: Is there a case when Branch Name will be null?

File: lazy-scripts/scan.sh [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+
+set -euo pipefail
+
+SONAR_ORGANIZATION=""
+SONAR_PROJECT_KEY=""
+REVISION=""
+BRANCH_NAME=""
+PULL_REQUEST_KEY=""
+
+function process_cli
```
Comment (vibindaniel) [inline lazy-scripts/scan.sh:11]: Tech debt ticket to make scripts consistant

**Review by vibindaniel — CHANGES_REQUESTED** [review]
Almost done

File: lazy-scripts/scan.sh (hunk 15 above)
Comment (naveen-variant) [reply lazy-scripts/scan.sh:11]: Ticket created to track in next sprint.

File: README.md (hunk 13 above)
Comment (naveen-variant) [reply README.md]: Updated Read me to show input parameters in tabular form.

File: lazy-scripts/run_tests_sonarscan.sh (hunk 14 above)
Comment (naveen-variant) [reply lazy-scripts/run_tests_sonarscan.sh:95]: Discussed in call . Processing differently based on the branch name .

File: lazy-scripts/build.sh (hunk 10 above)
Comment (naveen-variant) [reply lazy-scripts/build.sh:22]: Discussed in call . Referencing in nuget config.

File: entrypoint.sh (hunk 6 above)
Comment (naveen-variant) [reply entrypoint.sh]: Added comment

File: Dockerfile (hunk 7 above)
Comment (naveen-variant) [reply Dockerfile:22]: Updated to install awscli v2.

File: README.md [hunk 16]
```diff
@@ -1,2 +1,111 @@
...
+| `sonar_token`             |                       | Sonar token.Pass same secret as sample snippet.      | true     |                      |
+| `aws_access_key_id`       |                       | aws access key id.Pass same secret as sample snippet.| true     |                      |
+| `aws_secret_access_key`   |                       | aws secret access key.Same secret as sample snippet. | true     |                      |
+| `aws_region`              |                       | aws region.Pass same secret as sample snippet.       | false    |                      |
+| `docker_repo_name`        |                       | docker repository name.                              | true     |                      |
+| `docker_image_name`       |                       | docker image name.                                   | true     |                      |
+| `sonar_project_key`       |   `variant-inc`       | sonar project key.                                   | false    |                      |
+| `sonar_org`               |   `variant`           | sonar organization.                                  | false    |                      |
```
Comment (vibindaniel) [inline README.md:83]: default is `ariant` not variant

File: README.md [hunk 17]
```diff
@@ -1,2 +1,111 @@
...
+
+## parameters
+
+### Input Parameters
+
+
+
+| Parameter                 | Default               | Description                                          | Required |  Sample input        |
+|---------------------------|-----------------------|------------------------------------------------------|----------|----------------------|
+| `src_file_dir_path`       | `.`                   | Directory of the solution file                       | true     |'./' , './src/test'   |
```
Comment (vibindaniel) [inline README.md:72]: Don't use , in the sample. It is confusing. Someone can think that the parameter can support multiple entries separated by comma

File: README.md [hunk 18]
```diff
@@ -1,2 +1,111 @@
...
+| `dockerfile_dir_path`     | `.`                   | Directory of the dockerfile                          | true     |'./' , './src/test'   |
```
Comment (vibindaniel) [inline README.md:73]: same as above

File: README.md [hunk 19]
```diff
@@ -1,2 +1,111 @@
...
+| `aws_region`              |                       | aws region.Pass same secret as sample snippet.       | false    |                      |
+| `docker_repo_name`        |                       | docker repository name.                              | true     |                      |
+| `docker_image_name`       |                       | docker image name.                                   | true     |                      |
+| `sonar_project_key`       |   `variant-inc`       | sonar project key.                                   | false    |                      |
+| `sonar_org`               |   `variant`           | sonar organization.                                  | false    |                      |
+| `sonar_scan_enabled`      |    `false`            | set to true if sonar scan enabled or set to false.   | false    |                      |
+| `nuget_push`              |    `false`            | set to true if nuget push is enabled or set to false.| false    |                      |
+| `docker_push`             |    `false`            | set to true if push to ECR is enabled set to false.  | false    |                      |
```
Comment (vibindaniel) [inline README.md:86]: I'd default this to `true`

File: README.md [hunk 20]
```diff
@@ -1,2 +1,111 @@
...
+
+
+### Output Parameters
+
+Output parameters section.
+
+```text
```
Comment (vibindaniel) [inline README.md:93]: Please create a table as above

File: README.md [hunk 21]
```diff
@@ -1,2 +1,111 @@
...
+
+```yaml
+
+    - name: Get the image name from lazy-action
+      run: echo "Image Name:${{ steps.lazy-action.outputs.image }}"
+
+```
+steps.lazy-action.outputs.image :
+
+* lazy-action : id from your lazy action invocation.
```
Comment (vibindaniel) [inline README.md:110]: What is this ID for?

File: lazy-scripts/scan.sh (hunk 15 above)
Comment (vibindaniel) [reply lazy-scripts/scan.sh:11]: Next time, please share the link to the ticket

File: lazy-scripts/Dockerfile [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+
+# Copy csproj and restore as distinct layers
+COPY *.sln .
+COPY *.csproj ./
+RUN dotnet restore
+
+FROM build-env AS test-env
+RUN apk add --no-cache tzdata
+RUN ["dotnet", "test", "--no-build", "--no-restore"]
+CMD ["dotnet", "test", "--no-build", "--no-restore"]
```
Comment (vibindaniel) [inline lazy-scripts/Dockerfile:17]: CMD is not required

File: entrypoint.sh [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+fi
+
+echo "Start: Publish nuget push."
+echo "nugetflag: $INPUT_NUGET_PUSH"
+if [ $INPUT_NUGET_PUSH == 'true' ]
+then
+    echo "Start publishing image to github registry."
+    if [ -z "${INPUT_DOCKERFILE_DIR_PATH}" ];
+    then
+        echo "lazy flow" #todo
```
Comment (vibindaniel) [inline entrypoint.sh:68]: Please create a ticket for the same

**Review by vibindaniel — CHANGES_REQUESTED** [review]

File: lazy-scripts/run_tests_sonarscan.sh (hunk 12 above)
Comment (naveen-variant) [reply lazy-scripts/run_tests_sonarscan.sh:32]: https://usxtech.atlassian.net/browse/LAZY-51

File: lazy-scripts/scan.sh (hunk 15 above)
Comment (naveen-variant) [reply lazy-scripts/scan.sh:11]: https://usxtech.atlassian.net/browse/LAZY-51

File: README.md (hunk 16 above)
Comment (naveen-variant) [reply README.md:83]: Updated.

File: README.md (hunk 21 above)
Comment (naveen-variant) [reply README.md:110]: Just referencing id from the dotnet action steps to get the output prameter.

File: entrypoint.sh (hunk 23 above)
Comment (naveen-variant) [reply entrypoint.sh:68]: https://usxtech.atlassian.net/browse/LAZY-52

File: README.md (hunk 19 above)
Comment (naveen-variant) [reply README.md:86]: updated default as true.

File: README.md (hunk 20 above)
Comment (naveen-variant) [reply README.md:93]: table created for output paramters

File: lazy-scripts/Dockerfile (hunk 22 above)
Comment (naveen-variant) [reply lazy-scripts/Dockerfile:17]: Removed

File: README.md (hunk 17 above)
Comment (naveen-variant) [reply README.md:72]: Removed sample inputs.

**Review by vibindaniel — APPROVED** [review]

**Pull request merged**